### PR TITLE
[AKS] support `--yes` for `az aks mesh upgrade` commands.

### PR DESCRIPTION
**Pull request by deveshdama**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks mesh upgrade rollback -g myResourceGroup -n myCluster --revision asm-1-20`
`az aks mesh upgrade complete -g myResourceGroup -n myCluster`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 +++++++
+* support `--yes` for `az aks mesh upgrade rollback` command.
```
Comment (nshankar13) [inline src/aks-preview/HISTORY.rst]: Update this to also reflect `complete.`

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -3026,26 +3026,30 @@ def aks_mesh_upgrade_complete(
         cmd,
         client,
         resource_group_name,
-        name):
+        name,
+        yes):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:3030]: If yes is not a required parameter, please set a default value for it.

File: src/aks-preview/azext_aks_preview/custom.py [hunk 3]
```diff
@@ -3026,26 +3026,30 @@ def aks_mesh_upgrade_complete(
...
         mesh_upgrade_command=CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE)
 
 
 def aks_mesh_upgrade_rollback(
         cmd,
         client,
         resource_group_name,
-        name
+        name,
+        yes
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:3045]: same here

File: src/aks-preview/azext_aks_preview/_params.py [hunk 4]
```diff
@@ -2135,6 +2135,24 @@ def load_arguments(self, _):
             "revision", validator=validate_azure_service_mesh_revision, required=True
         )
 
+    with self.argument_context("aks mesh upgrade rollback") as c:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:2138]:
```suggestion
    with self.argument_context("aks mesh upgrade") as c:
```
could merge these 2 blocks, cli framework will match the sub commands with the same prefix

Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/_params.py:2138]: we do have `az aks mesh upgrade start` command as well which doesn't define `--yes` as part of it.

**Review by FumingZhang — COMMENTED** [review]
lgtm

But I find there's no scenario test for command `aks mesh upgrade rollback/complete`, a passed live test could help ensure that everything is working properly.

File: src/aks-preview/azext_aks_preview/_params.py [hunk 5]
```diff
@@ -2135,6 +2135,26 @@ def load_arguments(self, _):
...
+        c.argument(
+            "yes",
+            options_list=["--yes", "-y"],
+            help="Do not prompt for confirmation.",
+            action="store_true",
+            default=False,
+            required=False
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:2145]: I think you may omit `required=False` as this is the default behavior

File: src/aks-preview/azext_aks_preview/_params.py [hunk 6]
```diff
@@ -2135,6 +2135,26 @@ def load_arguments(self, _):
...
+        )
+
+    with self.argument_context("aks mesh upgrade complete") as c:
+        c.argument(
+            "yes",
+            options_list=["--yes", "-y"],
+            help="Do not prompt for confirmation.",
+            action="store_true",
+            default=False,
+            required=False
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:2155]: I think you may omit `required=False` as this is the default behavior

**Review by FumingZhang — APPROVED** [review]
[Live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=91708424&view=results) passed
- test_aks_azure_service_mesh_canary_upgrade

File: src/aks-preview/azext_aks_preview/_params.py (hunk 5 above)
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/_params.py]: I am running into this error on removing `required=False`
`the following arguments are required: --yes/-y`

File: src/aks-preview/azext_aks_preview/_params.py (hunk 6 above)
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/_params.py]: I am running into this error on removing required=False
the following arguments are required: --yes/-y

File: src/aks-preview/azext_aks_preview/_params.py (hunk 5 above)
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/_params.py:2145]: OK, if you set the default value to `False` in custom.py then you could probably remove the `default=False,
required=False` here.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 7]
```diff
@@ -2152,9 +2153,10 @@ def _handle_upgrade_asm(self, new_profile: ServiceMeshProfile) -> Tuple[ServiceM
                     f"Please ensure all data plane workloads have been rolled over to revision {revision_to_keep} "
                     "so that they are still part of the mesh.\nAre you sure you want to proceed?"
                 )
-                if prompt_y_n(msg, default="y"):
-                    new_profile.istio.revisions.remove(revision_to_remove)
-                    updated = True
+                if not supress_confirmation and not prompt_y_n(msg, default="n"):
```
Comment (SanyaKochhar) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2156]: I'm not sure I follow this line?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 8]
```diff
@@ -11920,6 +11926,125 @@ def test_aks_azure_service_mesh_with_egress_gateway(
...
+            upgrade_start_cmd,
+            checks=[
+                self.check("provisioningState", "Succeeded"),
+                self.check("serviceMeshProfile.mode", "Istio"),
+            ],
+        )
+
+        # upgrade complete
+        upgrade_complete_cmd = (
+            "aks mesh upgrade complete --resource-group={resource_group} --name={name} --yes"
```
Comment (SanyaKochhar) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:12024]: do we have/need to have any testing for when the --yes is not provided inline and is provided as a followup prompt?

Comment (SanyaKochhar) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:12024]: can follow precedent from other commands that use this option

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 9]
```diff
@@ -11920,6 +11926,125 @@ def test_aks_azure_service_mesh_with_egress_gateway(
...
+        """This test case exercises canary upgrade with mesh upgrade command.
+
+        It creates a cluster, enables azure service mesh, fetches available upgrade revison, upgrades the cluster then disable it.
+        """
+
+        # reset the count so in replay mode the random names will start with 0
+        self.test_resources_count = 0
+        # kwargs for string formatting
+        aks_name = self.create_random_name("cliakstest", 16)
+        installed_revision = self._get_asm_supported_revision(resource_group_location)
```
[the remark this answers is not in the document]
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:11945]: it takes the first revision returned by command `aks mesh get-revisions -l {location}`. I believe that we always atleast have two revisions returned by `get-revisions`. If this picks the latest and there's no upgrade available, runner test would start to fail and we would be alerted but I don't expect this to happen.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 8 above)
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:12024]: we need to have `--yes` arg in this command else the live test is stuck with the prompt and eventually fails. All other commands provide `--yes` as well where applicable.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 7 above)
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2156]: `prompt_y_n(..)` is only run if `supress_confirmation` is false and we proceed based on `prompt_y_n(..)` return value.
if `supress_confirmation` is true, we skip this if statement.

File: src/aks-preview/HISTORY.rst [hunk 10]
```diff
@@ -11,6 +11,10 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 +++++++
+
+3.0.0b6
++++++++
+* support `--yes` for `az aks mesh upgrade rollback` and `az aks mesh upgrade complete` commands.
```
Comment (yanzhudd) [inline src/aks-preview/HISTORY.rst:17]:
```suggestion
* Support `--yes` for `az aks mesh upgrade rollback` and `az aks mesh upgrade complete` commands.
```

File: src/aks-preview/HISTORY.rst [hunk 11]
```diff
@@ -11,6 +11,10 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 +++++++
+
+3.0.0b6
```
Comment (yanzhudd) [inline src/aks-preview/HISTORY.rst:15]: It seems that 3.0.0b6 has been released, please use a new version.
https://github.com/Azure/azure-cli-extensions/blob/c9e19421a81ff560878997156cfc21a40c8be0a2/src/aks-preview/setup.py#L12

File: src/aks-preview/setup.py [hunk 12]
```diff
@@ -9,7 +9,7 @@
 
 from setuptools import setup, find_packages
 
-VERSION = "3.0.0b5"
+VERSION = "3.0.0b6"
```
Comment (yanzhudd) [inline src/aks-preview/setup.py:12]: The 3.0.0b6 has been released, please update it to a new version.

**Review by yanzhudd — APPROVED** [review]

**Comment by FumingZhang** [comment]
CI failure is not related to the change in this PR, would be fixed in #7548.
```
>               self.agentpool.windows_profile.disable_windows_outbound_nat is not None
            ):
E           AttributeError: 'AgentPoolWindowsProfile' object has no attribute 'disable_windows_outbound_nat'
```

**Comment by yanzhudd** [comment]
Please resolve the code conflict.

**Comment by deveshdama** [comment]
resolved.

**Review by FumingZhang — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**